### DISCUSSION TITLE
Async action not wait for change

### DISCUSSION THREAD

**Discussion by Merynek** [mobxjs/mobx#2704] [Q&A]
Hello,

I have problem with mobx action with async method depend on changes.
I tried  lot of stuff for fix this but without success.
```
    @action
    private async processMethod(arg1: number, arg2: number) {
        this.bigStore.set1(arg1);  // this line change some observable properties => need rerender
        this.bigStore.set2(arg2);  // this line change some observable properties => need rerender
        // in this time rerender is called first time (OK)
        await this.processBigChange(); // this line is depend lines above and change lot of observable properties
        // in this time rerender is called second time (FAIL =>  need just one rerender)
    }
```
Have you ever had this issue?

**Version**
"mobx": "^5.15.7",
"mobx-react": "^6.3.1"

**Comment by mweststrate** [comment]
This is all documented. See the action docs, the section about writing
async actions.

**Reply by Merynek** [reply]
I saw that, but it is about easy async methods in action, not this case

**Reply by Merynek** [reply]
I looked to documentation and this case is not there. Ofc this is async method but it depends on changes before (observables) and this case I did not find there. :-(
